sdk/state/asset_test.go

```diff
@@ -0,0 +1,34 @@
+package state_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stellar/experimental-payment-channels/sdk/state"
+	"github.com/stellar/go/txnbuild"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAsset(t *testing.T) {
+	testCases := []struct {
+		Asset             state.Asset
+		WantTxnbuildAsset txnbuild.Asset
+		WantNative        bool
+		WantCode          string
+		WantIssuer        string
+	}{
+		{state.Asset(""), txnbuild.NativeAsset{}, true, "", ""},
+		{state.Asset("native"), txnbuild.NativeAsset{}, true, "", ""},
+		{state.NativeAsset, txnbuild.NativeAsset{}, true, "", ""},
+		{state.Asset(":"), txnbuild.CreditAsset{}, false, "", ""},
+		{state.Asset("ABCD:GABCD"), txnbuild.CreditAsset{Code: "ABCD", Issuer: "GABCD"}, false, "ABCD", "GABCD"},
```

Comment:
can we add a test case for something like `ABCD:GABCD:AB`?

Reply:
👍🏻 f140df3

Reply:
I added back a test I deleted and added an additional test in 9933490.